compiler/rustc_resolve/src/lib.rs

```diff
@@ -3337,35 +3337,20 @@ impl<'a> Resolver<'a> {
     // FIXME(Manishearth): intra-doc links won't get warned of epoch changes.
     pub fn resolve_str_path_error(
         &mut self,
-        span: Span,
         path_str: &str,
         ns: Namespace,
         module_id: DefId,
-    ) -> Result<(ast::Path, Res), ()> {
-        let path = if path_str.starts_with("::") {
-            ast::Path {
-                span,
-                segments: iter::once(Ident::with_dummy_span(kw::PathRoot))
-                    .chain(path_str.split("::").skip(1).map(Ident::from_str))
-                    .map(|i| self.new_ast_path_segment(i))
-                    .collect(),
-                tokens: None,
-            }
-        } else {
-            ast::Path {
-                span,
-                segments: path_str
-                    .split("::")
-                    .map(Ident::from_str)
-                    .map(|i| self.new_ast_path_segment(i))
-                    .collect(),
-                tokens: None,
-            }
-        };
+    ) -> Result<Res, ()> {
+        let mut segments = Vec::from_iter(
+            path_str.split("::").map(Ident::from_str).map(|i| self.new_ast_path_segment(i)),
+        );
+        if path_str.starts_with("::") {
+            segments[0].ident.name = kw::PathRoot;
+        }
+        let path = ast::Path { span: DUMMY_SP, segments, tokens: None };
```

Comment:
Why does this now use `DUMMY_SP` unconditionally? I remember some discussion in https://github.com/rust-lang/rust/issues/78696 that it would be necessary to resolve `crate::` properly.

Reply:
This is addressed in https://github.com/rust-lang/rust/pull/95337 without using the span.

Also if spans are used as an input, then they'd have to be a part of keys in the `doc_link_resolutions` table too.
Maybe we'll have to do it some time later to better support macro 2.0 hygiene in doc links, but now I'd prefer to omit it for simplicity.